app/controllers/topics_controller.rb

```diff
@@ -89,7 +89,7 @@ def show
     end
 
     if opts[:print]
-      raise Discourse::InvalidAccess unless SiteSetting.max_prints_per_hour_per_user > 0
+      raise Discourse::InvalidAccess if SiteSetting.max_prints_per_hour_per_user <= 0
```

Comment:
We could set a `min: 0` on this site setting to simplify this, anything less than 0 is not valid (0 is disabled in this case). Then we can do `raise Discourse::InvalidAccess if SiteSetting.max_prints_per_hour_per_user.zero?`